File: Napakalaki JAVA : GUI/src/napakalaki_java/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki_java;

/**
 *
 * @author alvarillo89
 */
public class CultistPlayer extends Player{
    private static int totalCultistPlayers = 0;
    private Cultist myCultistCard;
    
    public CultistPlayer(Player p, Cultist c){
        super(p);
        myCultistCard = c;
        totalCultistPlayers++;
    }
    
    @Override
    public int getCombatLevel(){
        int CombatLevel = super.getCombatLevel();
        double combatLv = CombatLevel * 0.7;
        CombatLevel = (int)combatLv;
        CombatLevel += myCultistCard.getGainedLevels();
        CombatLevel *= totalCultistPlayers;
        return CombatLevel;
    }
    
    @Override
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    
    @Override
    protected boolean shouldConvert(){
        return false;
    }
    
    @Override
    protected Treasure giveMeATreasure(){
        int numeroAleatorio;
        numeroAleatorio = (int) (Math.random()*(getVisibleTreasures().size()));
        Treasure t = getVisibleTreasures().get(numeroAleatorio);
        getVisibleTreasures().remove(t);
        
        return t;
    }
    
    @Override    
    protected boolean canYouGiveMeATreasure(){
        if(getVisibleTreasures().size() > 0)
            return true;
        
        return false;
    }
    
    public static int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
    
    public Cultist getCultistCard(){
        return myCultistCard;
    }
    
    @Override
    public String toString(){
        String s = "";
        s = s + getName() + " (Nivel: " + getLevel() + " Combat Level: " + getCombatLevel() + " Enemigo: " + enemy.getName() + "\nCultistCard = " + myCultistCard.toString() + ")";
        if(getPendingBad() != null)
            s = s + "\nPendingBadConsequence = " + getPendingBad().toString();
            
        return s;
    }
}

File: Napakalaki RUBY/lib/cultist_player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "napakalaki.rb"

module NapakalakiGame
class CultistPlayer < Player
  @@totalCultistPlayer = 0
  
  def initialize(p, c)
    super(p.getName(),1,nil,Array.new,Array.new,nil)
    copyPlayer(p)
    @myCultistCard = c
    @@totalCultistPlayer = @@totalCultistPlayer + 1
  end
  
  def self.getTotalCultistPlayers()
    @@totalCultistPlayer
  end
  
  private
  def giveMeATreasure()
    numeroAleatorio = rand(0..(Player.getVisibleTreasures().size))
    t = Player.getVisibleTreasures()[numeroAleatorio]
    Player.discardVisibleTreasure(t)
    return t
  end
  
  def canYouGiveMeATreasure()
    if(Player.getVisibleTreasures().size > 0)
      return true
    end
        
    return false
  end
  
  protected
    
  def getCombatLevel()
    level = super
    level = level + 0.7*level
    level = level + @myCultistCard.getGainedLevels()*@@totalCultistPlayer
    level = level.to_i
    
    return level
  end
  
  def getOponentLevel(m)
    m.getCombatLevelAgainstCultistPlayer()
  end
  
  def shouldConvert()
    return false
  end
end
end
